### ISSUE TITLE
Dependencies not aligned with 3D Slicer

### ISSUE THREAD

**Issue by jamesobutler** [PlusToolkit/PlusLib#1004]
3D Slicer 5.0.0 uses ITK 5.3rc03 and soon to be release Slicer 5.2.0 uses ITK 5.3rc04. PlusBuild is still specifying ITK 5.2.0 which results in the following warnings about potential incompatibility when connecting a PlusServer to Slicer (see output below). Generally PlusToolkit has tried to stay aligned with 3D Slicer regarding external toolkits such as ITK as previously discussed in https://github.com/PlusToolkit/PlusBuild/pull/52.

```
WARNING: In C:\SP\SP22R\itk\Modules\Core\Common\src\itkObjectFactoryBase.cxx, line 597
Possible incompatible factory load:
Running itk version :
itk version 5.2.0
Loaded factory version:
itk version 5.3.0
Loading factory:
C:/Program Files/SonoEQ 2.0.0-2022-10-14/bin/../lib/SonoEQ-5.1/ITKFactories\MRMLIDIOPlugin.dll
```

ITK 5.3 requires C++14 at minimum and an updated CMake minimum version.
- https://github.com/Slicer/Slicer/commit/bc2b3f4049ef32486cf43ad16903820fc8767f27
- https://github.com/Slicer/Slicer/commit/664fecaf8d70acdfaba30a473fe35931a4999e47

3D Slicer further moved to C++17 in https://github.com/Slicer/Slicer/commit/dedb6c04d5f987d168e58dfda067717a7039b439. I see that some devices in Plus such as ClariusOEM require C++17 as seen [here](https://github.com/PlusToolkit/PlusBuild/blob/master/CMakeLists.txt#L16-L23).

Due to this C++ minimum standard upgrade I have also observed issues building OpenCV where PlusBuild [specifies version 3.4.7](https://github.com/PlusToolkit/PlusBuild/blob/master/SuperBuild/External_OpenCV.cmake) which might be too old for a new C++ standard. The SlicerOpenCV extension builds and uses OpenCV [4.5.5](https://github.com/Slicer/SlicerOpenCV/blob/3f0bbd87c2e821020944d805405fe603874477c1/SuperBuild/External_OpenCV.cmake#L35).

@adamrankin, @Sunderlandkyl, @lassoan How should we proceed here in PlusToolkit with the updating of the dependencies to match 3D Slicer? Are we still going to try and maintain the use of old versions of these toolkits where some devices are requiring old versions and old C++ standards and some devices are requiring new versions and new standards?

**Comment by adamrankin** [comment]
Is the only device with an issue moving forward the StealthLink?

**Comment by jamesobutler** [comment]
It appears that StealthLink may be the reason for a lot of the old support.

I know that StealthLink was the reason that ITK 4 support was kept around. 

https://github.com/PlusToolkit/PlusBuild/commit/382fad1280cadd506f98a252bf0baf4a68e69740

> For StealthLink compatibility, Visual Studio 2013 is currently required.
> ITK 5.0.0 is still the default option, and configuring Plus with ITK 4.12.0 displays a warning to dissuade users from choosing it.

https://github.com/PlusToolkit/PlusBuild/commit/6a3da695b32d69fe3116f8f9cfa3f5ca86f7b326

> ITK_CXX_OPTIMIZATION_FLAGS and ITK_C_OPTIMIZATION_FLAGS can be set by the build environment to optimize performance, however this can cause the executable to not function on older CPU.
> 
> This commit sets the ITK optimization flags to an empty string to prevent optimization.


Old MSVC support checked at https://github.com/PlusToolkit/PlusBuild/blob/937a130bb4ea3aae90949842a41b20b4f04b2cc6/CMakeLists.txt#L51-L56

Old Qt WebKitWidgets supported only with Qt <5.6
https://github.com/PlusToolkit/PlusBuild/blob/937a130bb4ea3aae90949842a41b20b4f04b2cc6/CMakeLists.txt#L475-L479

**Comment by adamrankin** [comment]
@lassoan @Sunderlandkyl Maybe we make one final StealthLink Plus version, and remove support going forward?

**Comment by jamesobutler** [comment]
Would this final StealthLink version be a released PlusLib 2.10.0, or just a tagged commit? Would bumping of the required C++ version to correspond with latest ITK and so forth warrant a bump to PlusLib 3.x development?

**Comment by jamesobutler** [comment]
- PlusBuild now requires C++14 for non-stealthlink builds (see https://github.com/PlusToolkit/PlusBuild/commit/3d0a82c380f50428d53c13d181a22a209cb46a0d)
- PlusBuild now specifies ITK 5.3.0 (see https://github.com/PlusToolkit/PlusBuild/commit/10e7d458517c73ffa21a1a4f99fea376a07f8cc8)
